test/quantization/fx/test_model_report_fx.py

```diff
@@ -1146,15 +1146,58 @@ def test_qconfig_mapping_generation(self):
         Tests for generation of qconfigs by ModelReport API
         - Tests that qconfigmapping is generated
         - Tests that mappings include information for for relavent modules
+        - Edge case with no detectors
+        - Errors if tried before report generation
         """
-        pass
+        with override_quantized_engine('fbgemm'):
+            # set the backend for this test
+            torch.backends.quantized.engine = "fbgemm"
+            # test with multiple detectors
+            detector_set = set()
+            detector_set.add(OutlierDetector(reference_percentile=0.95))
+            detector_set.add(DynamicStaticDetector())
+
+            model = TwoThreeOps()
+
+            # get tst model and callibrate
+            prepared_for_callibrate_model, mod_report = _get_prepped_for_calibration_model_helper(
+                model, detector_set, model.get_example_inputs()[0]
+            )
+
+            # now we actually callibrate the models
+            example_input = model.get_example_inputs()[0]
+            example_input = example_input.to(torch.float)
+
+            prepared_for_callibrate_model(example_input)
+
+            # try to generate mapping without generating report, should throw error
+            # only really need to test one of them here
+            with self.assertRaises(Exception):
+                qconfig_mapping = mod_report.generate_qconfig_mapping()
```

Comment:
is it possible to generate the qconfig_mapping from the report object? so that we don't need to do these checks